### PR TITLE
Refactoring the depth estimation (DAv3) inference code

### PR DESCRIPTION
**Pull request by mrpositron**
## What has changed and why?

- Remove PIL, OpenCV, and numpy from the DAv3 inference transforms; preprocessing is now pure PyTorch in `image_utils.py` (the `InputProcessor` class is removed).
  - OpenCV's `INTER_AREA` and `INTER_CUBIC` resizes are reimplemented in PyTorch and match `cv2.resize` bit-exactly.
- `predict()` now returns the depth map at the original input resolution instead of the DA3 processing resolution.
- Add `predict_batch()` for mixed-size images: processed images are center-cropped to the smallest size in the batch, and each depth map is resized back to its original resolution.

## How has it been tested?
- New unit tests for `image_utils.py`: parity with `cv2.resize` on random images, plus hand-computed edge cases. 
- New task model tests: `predict` and `predict_batch` return depth maps at the original input resolution; `forward` returns depth and sky maps.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

**Comment by mrpositron** [comment]
@codex

File: src/lightly_train/_task_models/depth_estimation_components/image_utils.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+
+def _unify_sizes(images: list[Tensor]) -> list[Tensor]:
+    """Center-crops all images to the smallest height and width in the batch."""
+    sizes = {tuple(img.shape[-2:]) for img in images}
+    if len(sizes) <= 1:
+        return images
+    min_h = min(h for h, _ in sizes)
+    min_w = min(w for _, w in sizes)
+    logger.warning(
```
Comment (liopeer) [inline src/lightly_train/_task_models/depth_estimation_components/image_utils.py:100]: Could also imagine just an info-level message here. But also fine with a warning.

**Review by liopeer — APPROVED** [review]
LGTM! Feel free to resolve on your own.
